syntax = "proto3";
package spellfire;
 
import "types.proto";
 
option go_package = "github.com/Icemaster-Eric/Spellfire/backend/internal/proto;pb";
 
message ClientEvent {
  enum ClientEventType {
    CLIENT_EVENT_TYPE_UNSPECIFIED = 0;
    MOVE = 1;
    START_FIRE = 2;
    STOP_FIRE = 3;
  }
  ClientEventType type = 1;
  Vec2 movement = 2;
}
 
message ClientPacket {
  Timestamp timestamp = 1;
  repeated ClientEvent events = 2;
}